layers/core_validation.cpp

```diff
@@ -18314,12 +18303,263 @@ bool CoreChecks::ValidateCreateSwapchain(const char *func_name, VkSwapchainCreat
+                             "), minImageExtent = (%" PRIu32 ",%" PRIu32 "), "
+                             "maxImageExtent = (%" PRIu32 ",%" PRIu32 ").",
+                             func_name, pCreateInfo->imageExtent.width, pCreateInfo->imageExtent.height,
+                             capabilities.currentExtent.width, capabilities.currentExtent.height, capabilities.minImageExtent.width,
+                             capabilities.minImageExtent.height, capabilities.maxImageExtent.width,
+                             capabilities.maxImageExtent.height)) {
+                    return true;
+                }
+            }
+        }
+    } else {
+        // Validate VkSwapchainPresentModesCreateInfoEXT data
+        auto swapchain_present_modes_ci = LvlFindInChain<VkSwapchainPresentModesCreateInfoEXT>(pCreateInfo->pNext);
+        if (swapchain_present_modes_ci) {
+            bool found_swapchain_modes_ci_present_mode = false;
+            for (uint32_t i = 0; i < swapchain_present_modes_ci->presentModeCount; i++) {
+                VkPresentModeKHR swapchain_present_mode = swapchain_present_modes_ci->pPresentModes[i];
+
+                bool found_match =
+                    std::find(present_modes.begin(), present_modes.end(), swapchain_present_mode) != present_modes.end();
+                if (!found_match) {
+                    if (LogError(device, "VUID-VkSwapchainPresentModesCreateInfoEXT-None-07762",
+                                 "%s was called with vkSwapchainPresentModesCreateInfoEXT structure in its pNext chain that "
+                                 "contains a non-supported presentMode in pPresentModes[%d]: (%s).",
+                                 func_name, i, string_VkPresentModeKHR(swapchain_present_mode))) {
+                        skip |= true;
+                    }
+                }
+
+                std::vector<VkPresentModeKHR> compatible_present_modes =
+                    surface_state->GetCompatibleModes(physical_device, present_mode);
+                found_match = std::find(compatible_present_modes.begin(), compatible_present_modes.end(), swapchain_present_mode) !=
+                              compatible_present_modes.end();
+                if (!found_match) {
+                    if (LogError(device, "VUID-VkSwapchainPresentModesCreateInfoEXT-pPresentModes-07763",
+                                 "%s was called with vkSwapchainPresentModesCreateInfoEXT structure in its pNext chain that "
+                                 "contains a non-compatible presentMode in pPresentModes[%d]: (%s).",
+                                 func_name, i, string_VkPresentModeKHR(swapchain_present_mode))) {
+                        skip |= true;
+                    }
+                }
+
+                if (swapchain_present_modes_ci->pPresentModes[i] == present_mode) {
+                    found_swapchain_modes_ci_present_mode = true;
+                }
```

Comment:
Avoid if statement with `|=`

```suggestion
                const bool has_present_mode = (swapchain_present_modes_ci->pPresentModes[i] == present_mode);
                found_swapchain_modes_ci_present_mode |= has_present_mode;
```

Reply:
Done.  Uncomfortable, but done.